src/coreclr/src/tools/crossgen2/ILCompiler.ReadyToRun/Compiler/ReadyToRunTableManager.cs

```diff
@@ -54,22 +77,97 @@ protected virtual void Graph_NewMarkedNode(DependencyNodeCore<NodeFactory> obj)
             {
                 if (!_sortedMethods)
                 {
                     TypeSystemComparer comparer = new TypeSystemComparer();
-                    _methodsGenerated.Sort((x, y) => comparer.Compare(x, y));
+                    Comparison<IMethodNode> sortHelper = (x, y) => comparer.Compare(x.Method, y.Method);
+
+                    List<PerModuleMethodsGenerated> perModuleDatas = new List<PerModuleMethodsGenerated>(_methodsGenerated.Values);
+                    perModuleDatas.Sort((x, y) => x.Module.CompareTo(y.Module));
+
+                    foreach (var perModuleData in perModuleDatas)
+                    {
+                        perModuleData.MethodsGenerated.Sort(sortHelper);
+                        perModuleData.GenericMethodsGenerated.Sort(sortHelper);
+                        _completeSortedMethods.AddRange(perModuleData.MethodsGenerated);
+                        _completeSortedMethods.AddRange(perModuleData.GenericMethodsGenerated);
+                        _completeSortedGenericMethods.AddRange(perModuleData.GenericMethodsGenerated);
+                    }
                     _sortedMethods = true;
                 }
             }
-            return _methodsGenerated;
+            if (moduleToEnumerate == null)
+            {
+                if (methodCategory == CompiledMethodCategory.All)
+                {
+                    return _completeSortedMethods;
+                }
+                else if (methodCategory == CompiledMethodCategory.Instantiated)
+                {
+                    return _completeSortedGenericMethods;
+                }
+                else
+                {
+                    // This isn't expected to be needed, and thus isn't implemented
+                    throw new ArgumentException();
+                }
+            }
+            else if (_methodsGenerated.TryGetValue(moduleToEnumerate, out var perModuleData))
+            {
+                if (methodCategory == CompiledMethodCategory.All)
+                {
+                    return GetCompiledMethodsAllMethodsInModuleHelper(moduleToEnumerate);
+                }
+
+                if (methodCategory == CompiledMethodCategory.Instantiated)
+                {
+                    return perModuleData.GenericMethodsGenerated;
+                }
+                else
+                {
+                    Debug.Assert(methodCategory == CompiledMethodCategory.NonInstantiated);
+                    return perModuleData.MethodsGenerated;
+                }
+            }
+            else
+            {
+                return Array.Empty<IMethodNode>();
+            }
+        }
+
+        private IEnumerable<IMethodNode> GetCompiledMethodsAllMethodsInModuleHelper(EcmaModule moduleToEnumerate)
+        {
+            foreach (var node in GetCompiledMethods(moduleToEnumerate, CompiledMethodCategory.Instantiated))
+            {
+                yield return node;
+            }
```

Comment:
Would be nice if C# had an equivalent from Python's `yield from`, and being able to write this as `yield return from GetCompiledMethods(...)`.